container/docker/factory.go

```diff
@@ -42,6 +42,10 @@ var DockerNamespace = "docker"
 var dockerRootDir = flag.String("docker_root", "/var/lib/docker", "Absolute path to the Docker state root directory (default: /var/lib/docker)")
 var dockerRunDir = flag.String("docker_run", "/var/run/docker", "Absolute path to the Docker run directory (default: /var/run/docker)")
 
+// Regexp that identifies docker cgroups, containers started with
+// --cgroup-parent have another prefix than 'docker'
+var dockerCgroupRegexp = regexp.MustCompile(`[A-z]+-([a-z0-9]+)\.scope`)
```

Comment:
AFAIK format is `<cgroupParent>-<containerId>.scope`. Container ID is 64 characters (hence the `{64}` so from this regex we should be able to extract the cgroup parent (normally `docker`) & the container ID. I can't imagine any conflict with other scope units so this should be pretty safe.